proton-c/bindings/python/proton/utils.py

```diff
@@ -132,10 +132,12 @@ def __init__(self, connection, receiver, fetcher, credit=1):
             raise LinkException("Failed to open receiver %s, source does not match" % self.link.name)
         if credit: receiver.flow(credit)
         self.fetcher = fetcher
+        self.container = connection.container
 
     def __del__(self):
         self.fetcher = None
-        self.link.handler = None
+        self.link.handler = None  # fails if reactor finalizes first
```

Comment:
A core dump in pn_record_get.  That's why the container reference is set on init (line 135) and unset in the next line (line 140).  The order of decref to 0 differs for BlockingSenders and BlockingReceivers.  Sometimes the reactor is last, sometimes it precedes the blocking endpoints.  This bug lurked but was never hit because none of these objects were free'd before.